apps/speed.c

```diff
@@ -911,74 +919,251 @@ static int EVP_Update_loop(void *args)
  * CCM does not support streaming. For the purpose of performance measurement,
  * each message is encrypted using the same (key,iv)-pair. Do not use this
  * code in your application.
+ * For decryption, we will use buf2 to preserve the input text in buf.
  */
-static int EVP_Update_loop_ccm(void *args)
+static int EVP_Update_loop_ae(void *args)
```

Comment:
Since this function is being renamed and significantly rewritten, and the code between in the if/else is being significantly increased, would it be possible to split this into `EVP_Update_loop_ae_encrypt()` and `EVP_Update_loop_ae_decrypt()`?

Reply:
Thanks for the review Rom. Really appreciated. 

Sure, I guess it makes sense to have two separate functions rather than one giant function that does multiple things. I assume the same thing should be done to the EVP_Update_loop_aead() function as well right? i.e. split into encrypt/decrypt functions.

Reply:
> I assume the same thing should be done to the EVP_Update_loop_aead() function as well right?

Yes, that makes sense.

> Would it make sense to split these functions in another PR or shall we cancel this one and start working on a new one?

No need to cancel this PR - if you add another commit to your `evp-aead-fix` branch, it will get added to this PR - the changes are related (changing decrypt to have real data rather than random data is not a trivial change, so it's expected that there could be more than just one or two lines different, and refactoring functions that grow significantly is therefore also reasonable as part of the same PR).

And by the way: thank you for doing this work

Reply:
You're most welcome Tom; I'm very pleased to have been of help. I've just made a commit for splitting into encrypt/decrypt functions. also, we no longer have two separate functions `EVP_Update_loop_ae()` and `EVP_Update_loop_aead()` to handle AE and AEAD respectively. Now we only have `EVP_Update_loop_aead_enc()` and `EVP_Update_loop_aead_dec()`. They will process AAD only when the `-aead` flag is passed. Any further feedback is highly appreciated. Thanks